src/Components/SigDetailsTable/NewSigDetailsTable.js

```diff
@@ -107,23 +110,30 @@ const NewSigDetailsTable = ({ ruleName, affectedCount, isEmptyAccount }) => {
     ...initialState,
   });
 
+  const handleSetSearchParams = (displayName, hostWorkspace, status) => {
+    if (!displayName?.length) {
+      setSearchParams({ hostWorkspace, status });
+    } else if (!hostWorkspace.length) {
+      setSearchParams({ displayName, status });
+    } else if (!status.length) {
+      setSearchParams({ displayName, hostWorkspace });
+    } else {
+      setSearchParams({ displayName, hostWorkspace, status });
+    }
+  };
```

Comment:
Why is this so complex?

It's not very obvious why all these `!...length` are here.

Reply:
Because if `hostWorkspace` is `[]`, then I don't want to `setSearchParams({ displayName, hostWorkspace, status })`. I want to just `setSearchParams({ displayName, status })`